sensor/extrinsic_reflector_based_calibrator/scripts/metrics_plotter_node.py

```diff
@@ -0,0 +1,251 @@
+        self.subplot0 = self.axes[0, 0]
+        self.subplot1 = self.axes[0, 1]
+        self.subplot2 = self.axes[1, 0]
+        self.subplot3 = self.axes[1, 1]
+        plt.gcf().canvas.set_window_title("Metrics plotter")
+
+        self.color_distance_o = "C0o-"
+        self.color_yaw_o = "C1o-"
+        self.color_distance = "C0"
+        self.color_yaw = "C1"
+
+        (
+            self.num_of_reflectors_list,
+            self.calibration_distance_error_list,
+            self.calibration_yaw_error_list,
+            self.crossval_sample_list,
+            self.crossval_distance_error_list,
+            self.crossval_yaw_error_list,
+            self.std_crossval_distance_error_list,
+            self.std_crossval_yaw_error_list,
+        ) = ([], [], [], [], [], [], [], [])
+
+        self.m_to_cm = 100
+
+        self.plot_label_and_set_xy_lim()
+        plt.tight_layout()
+        plt.pause(0.1)
+
+    def plot_label_and_set_xy_lim(self):
+        self.subplot0.set_title("cross-validation error: distance")
+        self.subplot0.set_xlabel("number of tracks")
+        self.subplot0.set_ylabel("distance error [cm]")
+
+        self.subplot1.set_title("cross-validation error: yaw")
+        self.subplot1.set_xlabel("number of tracks")
+        self.subplot1.set_ylabel("yaw error [deg]")
+
+        self.subplot2.set_title("average error: distance")
+        self.subplot2.set_xlabel("number of tracks")
+        self.subplot2.set_ylabel("distance error [cm]")
+
+        self.subplot3.set_title("average error: yaw")
+        self.subplot3.set_xlabel("number of tracks")
+        self.subplot3.set_ylabel("yaw error [deg]")
+
+        max_ylim2 = (
+            max(self.calibration_distance_error_list) if self.calibration_distance_error_list else 5
+        )
+        max_ylim3 = max(self.calibration_yaw_error_list) if self.calibration_yaw_error_list else 1
+
+        cross_val_xlim = (
+            self.crossval_sample_list[-1]
+            if self.crossval_sample_list and self.crossval_sample_list[-1] >= 5
+            else 5
+        )
+        avg_xlim = (
+            self.num_of_reflectors_list[-1]
+            if self.num_of_reflectors_list and self.num_of_reflectors_list[-1] >= 5
+            else 5
+        )
+
+        self.subplot0.set_xlim(2.9, cross_val_xlim + 0.3)
+        self.subplot1.set_xlim(2.9, cross_val_xlim + 0.3)
+        self.subplot2.set_xlim(2.9, avg_xlim + 0.3)
+        self.subplot3.set_xlim(2.9, avg_xlim + 0.3)
+
+        self.subplot2.set_ylim(0, max_ylim2 + 5)
+        self.subplot3.set_ylim(0, max_ylim3 + 0.1)
+
+        for ax in self.axes.flat:
+            ax.xaxis.set_major_locator(plt.MaxNLocator(integer=True))
+
+    def is_delete_operation(self, msg_array):
+        if self.num_of_reflectors_list and msg_array[0] < self.num_of_reflectors_list[-1]:
+            return True
+        return False
+
+    def remove_avg_error_from_list(self):
+        if len(self.num_of_reflectors_list) == 1:
```

Comment:
How about i in range(min(2, len())) ? (That also takes the case where the len = 0)

Reply:
@vividf 
I think you misunderstood the meaning of this comment judging from what you did in
55f555fdbbeece094a52aaabf448e4baa6f9fc78
You should be able to replace the if/else with the code I provided

Reply:
@knzo25 Thanks! This is fixed with commit https://github.com/tier4/CalibrationTools/pull/145/commits/61ef8f75b8f77e57b0eb5b669354204bce44f405